### PR TITLE
Replace Newtonsoft.Json package with System.Text.Json

### PR DESCRIPTION
**Pull request by Bertk**
- uses System.Text.Json instead of Newtonsoft.Json for nuget packages
- integration test is still using Newtonsoft.Json
- generates code coverage report #1516 (Azure DevOps build pipeline)

:exclamation: PR #1510 should be merged before this one. The System.Text.Json updates shall be merges separately.

### CODE REVIEW

**Comment by Bertk** [comment]
@daveMueller @MarcoRossignoli 
This PR replaces **Newtonsoft.Json** from `coverlet.core` with **System.Text.Json** and removes many assemblies from the nuget packages. Please share your thoughts on this.

image

**Comment by MarcoRossignoli** [comment]
That's really cool, @daveMueller do you plan to review this one?

**Comment by daveMueller** [comment]
Yes I will. As I understood it's the same changes as in #1510 but instead of updating `Newtonsoft.Json`it uses `System.Text.Json`.

**Comment by Bertk** [comment]
I will rebase this branch after #1510 is merged :wink:

**Comment by MarcoRossignoli** [comment]
@Bertk I see you a lot active on coverlet, let me know if you're interested to become a co-maintainer and help to keep the product in a good shape.

You can contact me on twitter https://twitter.com/MarcoRossignoli or let me know how we can have a chat

Not mandatory at all it's OSS so no obligation.

cc: @daveMueller @petli @tonerdo

**Comment by Bertk** [comment]
@MarcoRossignoli Can we have a chat on Monday Zoom or MS Teams will be fine.

**Comment by Bertk** [comment]
@daveMueller @MarcoRossignoli Please review and approve PR.

Please note: I wanted to check stderr with the last commit but after the update the CI tests were successful.

By the way, the obsolete PR #1358 should be closed.

**Comment by Bertk** [comment]
@daveMueller Please review PR. This has also the code coverage report for the AzureDevops build pipeline.

File: eng/publish-coverage-results.yml
Comment (daveMueller) [inline eng/publish-coverage-results.yml:19]: If we need to restore the tool at every build anyway we could also just install it and wouldn't need `dotnet-tools.json`? Sth. like:
```powershell
dotnet tool install dotnet-reportgenerator-globaltool
dotnet reportgenerator -reports:"${{parameters.reports}}"
```
Unfortunately I can't really test anything related to the Azure DevOps pipeline thus I can't really confirm if it would work.

File: src/coverlet.core/CoverageResult.cs
Comment (daveMueller) [inline src/coverlet.core/CoverageResult.cs:13]: I think we can drop the `JsonInclude` attribute if we add the `IncludeFields` property to the json serializer of the `JsonReporter`.
```csharp
// JsonReporter
      var options = new JsonSerializerOptions
      {
        IncludeFields = true,
        WriteIndented = true
      };
```

File: src/coverlet.collector/coverlet.collector.csproj
Comment (Bertk) [reply src/coverlet.collector/coverlet.collector.csproj:39]: Missed this. Removed.

File: src/coverlet.core/coverlet.core.csproj
Comment (Bertk) [reply src/coverlet.core/coverlet.core.csproj:16]: This was required before `CentralPackageTransitivePinningEnabled` was used. Removed PackagerReference.

File: src/coverlet.core/Reporters/JsonReporter.cs
Comment (Bertk) [reply src/coverlet.core/Reporters/JsonReporter.cs:21]: Done.

File: Directory.Packages.props
Comment (Bertk) [reply Directory.Packages.props:49]: This will define the version which is used for the nuget package. I think we should use latest version.

File: test/coverlet.integration.tests/DotnetTool.cs
Comment (Bertk) [reply test/coverlet.integration.tests/DotnetTool.cs:38]: Correct. The error message will be available in the test result details (Azure DevOps, Visual Studio) and you see problems without checking txt or log files.

File: eng/publish-coverage-results.yml
Comment (Bertk) [reply eng/publish-coverage-results.yml:19]: My preferred solution is the Azure DevOps pipeline task (see #1516) because the installation and version update is done automatically. 
The local dotnet tool works in the build pipeline without additional permissions and version update can be done with **dependabot** .  `dotnet tool install -g ...` will fail if the tools was already installed by some other script.

File: src/coverlet.core/Reporters/JsonReporter.cs
Comment (Bertk) [reply src/coverlet.core/Reporters/JsonReporter.cs:5]: I activated option `JavaScriptEncoder.UnsafeRelaxedJsonEscaping`. We use only **XML file** or **console output** and not a **payload** for the coverage results.

>Unlike the [Default](https://learn.microsoft.com/en-us/dotnet/api/system.text.encodings.web.javascriptencoder.default?view=net-7.0#system-text-encodings-web-javascriptencoder-default) encoder, this encoder instance does not escape HTML-sensitive characters such as <, >, &. As a result, it must be used cautiously; for example, it can be used if the output data is within a response whose content-type is known with a charset set to UTF-8.

>:heavy_exclamation_mark: Compared to the default encoder, the UnsafeRelaxedJsonEscaping encoder is more permissive about allowing characters to pass through unescaped: 
>
>- It doesn't escape HTML-sensitive characters such as <, >, &, and '.
>- It doesn't offer any additional defense-in-depth protections against XSS or information disclosure attacks, such as those which might result from the client and server disagreeing on the charset.
>
>Use the unsafe encoder only when it's known that the client will be interpreting the resulting payload as UTF-8 encoded JSON. For example, you can use it if the server is sending the response header Content-Type: application/json; charset=utf-8. Never allow the raw UnsafeRelaxedJsonEscaping output to be emitted into an HTML page or a <script> element.

**Comment by Bertk** [comment]
@daveMueller I added coverage badge and trx files which have the output attachments. The PR is ready for merge.

**Review by daveMueller — APPROVED** [review]
Thanks @Bertk 👍

**Pull request merged**
